src/app/AlwaysOn.CatalogService/Controllers/CatalogItemController.cs

```diff
@@ -131,8 +131,18 @@ public async Task<ActionResult<CatalogItem>> CreateNewCatalogItemAsync(CatalogIt
 
             var itemId = itemDto.Id ?? Guid.NewGuid();
 
+            var newItem = new CatalogItem()
```

Comment:
The primary motivation was because EF needs to work with the fetched existing object to maintain tracking information, but then I realized too that there's no point in translating this to DTO, because the data comes from database.

Reply:
ok, cant remember if this would break anything, so I just trust you :)